### PR TITLE
Blazor WASM caching content reorganization

### LINKED ISSUE #28134: Document Blazor's caching behavior
Blazor does its own caching, which can catch developers by surprise if they try to disable browser caching but then find that their app is still cached. We mention Blazor's caching behavior indirectly in a couple of places, like here: https://learn.microsoft.com/aspnet/core/blazor/host-and-deploy/webassembly#resolve-integrity-check-failures. But it seems like it should get a more direct treatment.

@mkArtakMSFT  Is there someone on the team that can help out with content for this?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: ad891f84-d11e-e334-fcdc-b55c5171fca4
* Version Independent ID: 11f8db3d-9988-e0a1-89c5-c3552ee47b93
* Content: [ASP.NET Core Blazor startup](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-7.0)
* Content Source: [aspnetcore/blazor/fundamentals/startup.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/fundamentals/startup.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #28134

Looking good thus far ...

* Create a *WebAssembly caching* node under *Host and deploy*.
* Pull out the existing content into a new article (an overview) in the new node. The WASM article has grown too ***L....O....N....G*** 😩 anyway over the years, so it's nice to shorten it here.
* Move the HTTP caching article to the new node.

If this is heading the right direction thus far, there are at least two obvious questions ...

* Do we need to move any of the existing content around further?
* Are there additional introductory remarks to make on WASM caching in the overview article?

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/fundamentals/startup.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/blazor/fundamentals/startup.md) | [ASP.NET Core Blazor startup](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?branch=pr-en-us-32196) |
| [aspnetcore/blazor/host-and-deploy/webassembly-caching/http-caching-issues.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/blazor/host-and-deploy/webassembly-caching/http-caching-issues.md) | [aspnetcore/blazor/host-and-deploy/webassembly-caching/http-caching-issues](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly-caching/http-caching-issues?branch=pr-en-us-32196) |
| [aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md) | [aspnetcore/blazor/host-and-deploy/webassembly-caching/index](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly-caching/index?branch=pr-en-us-32196) |
| [aspnetcore/blazor/host-and-deploy/webassembly.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/blazor/host-and-deploy/webassembly.md) | [Host and deploy ASP.NET Core Blazor WebAssembly](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?branch=pr-en-us-32196) |
| [aspnetcore/blazor/progressive-web-app.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/blazor/progressive-web-app.md) | [[Visual Studio](#tab/visual-studio)](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/progressive-web-app?branch=pr-en-us-32196) |
| [aspnetcore/migration/70-80.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/migration/70-80.md) | [Migrate from ASP.NET Core 7.0 to 8.0](https://review.learn.microsoft.com/en-us/aspnet/core/migration/70-80?branch=pr-en-us-32196) |
| [aspnetcore/release-notes/aspnetcore-8.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/release-notes/aspnetcore-8.0.md) | [What's new in ASP.NET Core 8.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?branch=pr-en-us-32196) |
| [aspnetcore/whats-new/dotnet-AspNetCore.Docs-mod5.md](https://github.com/dotnet/AspNetCore.Docs/blob/548bd87fd0ada3d9fe3f7aea17561fd3668d6d16/aspnetcore/whats-new/dotnet-AspNetCore.Docs-mod5.md) | [aspnetcore/whats-new/dotnet-AspNetCore.Docs-mod5](https://review.learn.microsoft.com/en-us/aspnet/core/whats-new/dotnet-AspNetCore.Docs-mod5?branch=pr-en-us-32196) |

### CODE REVIEW

**Comment by guardrex** [comment]
@danroth27 ... Do you want to look at this?

File: aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+* Ensures the user's browser never caches inconsistent or invalid responses, which can prevent the app from starting even if the user manually refreshes the page.
+* Makes it safe to cache the responses and not check for server-side changes until the expected SHA-256 hashes themselves change, so subsequent page loads involve fewer requests and complete faster.
+
+If the web server returns responses that don't match the expected SHA-256 hashes, an error similar to the following example appears in the browser's developer console:
+
+> Failed to find a valid digest in the 'integrity' attribute for resource 'https://myapp.example.com/\_framework/MyBlazorApp.dll' with computed SHA-256 integrity 'IIa70iwvmEg5WiDV17OpQ5eCztNYqL186J56852RpJY='. The resource has been blocked.
+
+In most cases, the warning doesn't indicate a problem with integrity checking. Instead, the warning usually means that some other problem exists.
+
+For Blazor WebAssembly's boot reference source, see [the `Boot.WebAssembly.ts` file in the `dotnet/aspnetcore` GitHub repository](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web.JS/src/Boot.WebAssembly.ts).
```
Comment (danroth27) [inline aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md:37]: Most of the relevant code is actually in other places in the runtime code that we call in to. Probably best to just leave this out.

```suggestion
```

File: aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+When a Blazor WebAssembly app loads in the browser, the app downloads boot resources from the server:
+
+* JavaScript code to bootstrap the app
+* .NET runtime and assemblies
+* Locale specific data
+
+Except for Blazor's boot resources file (`blazor.boot.json`), WebAssembly .NET runtime and app bundle files are cached on clients.
```
Comment (danroth27) [inline aspnetcore/blazor/host-and-deploy/webassembly-caching/index.md:21]:
```suggestion
Except for Blazor's boot resources file (`blazor.boot.json`), WebAssembly .NET runtime and app bundle files are cached on clients. The `blazor.boot.json` file contains a manifest of all the files that make up the app that need to be download along with a hash of all the file's content that is used to detect whether any of the boot resources have changed.
```

**Review by danroth27 — APPROVED** [review]
A couple of minor comments but otherwise looks good! :+1:

**Pull request merged**